File: spigot/loader/src/main/java/com/wolfyscript/scafall/loader/SpigotLoaderPlugin.java
package com.wolfyscript.scafall.loader;

import org.bukkit.plugin.java.JavaPlugin;

class SpigotLoaderPlugin extends JavaPlugin {

    private final PluginBootstrap plugin;

    public SpigotLoaderPlugin() {
        ScaffoldingLoader.initAPIClassLoader(InnerJarClassloader.create(getClassLoader(), "scaffolding-api.innerjar"));

        var bootstrap = ScaffoldingLoader.loadScaffoldingBootstrap("scaffolding-spigot.innerjar");
        plugin = bootstrap.initScaffoldingPlatform("com.wolfyscript.scafall.spigot.ScaffoldingSpigotBootstrap", this);
    }

    @Override
    public void onLoad() {
        plugin.onLoad();
    }

    @Override
    public void onEnable() {
        plugin.onEnable();
    }

    @Override
    public void onDisable() {
        plugin.onUnload();
    }
}

File: spigot/loader/src/main/kotlin/com/wolfyscript/scaffolding/spigot/loader/SpigotLoaderPlugin.kt
package com.wolfyscript.scaffolding.spigot.loader

import com.wolfyscript.scaffolding.common.api.classloader.InnerJarClassLoader
import com.wolfyscript.scaffolding.common.bootstrap.PluginBootstrap
import org.bukkit.plugin.java.JavaPlugin

class SpigotLoaderPlugin : JavaPlugin() {

    private val plugin: PluginBootstrap

    init {
        val classLoader = InnerJarClassLoader(javaClass.classLoader, "scaffolding-spigot.innerjar")
        plugin = classLoader.loadPlugin("com.wolfyscript.scaffolding.spigot.ScaffoldingSpigot", this)
    }

    override fun onLoad() = plugin.onLoad()

    override fun onEnable() = plugin.onEnable()

    override fun onDisable() = plugin.onUnload()

}